src/lib/markbind/src/parser.js

```diff
@@ -95,17 +97,32 @@ class Parser {
     $('variable').each(function () {
       const variableElement = $(this);
       const variableName = variableElement.attr('name');
-      if (!variableName) {
-        // eslint-disable-next-line no-console
-        console.warn(`Missing 'name' for variable in ${fileName}\n`);
-        return;
-      }
-      if (!pageVariables[variableName]) {
-        const variableValue = nunjuckUtils.renderEscaped(nunjucks, md.renderInline(variableElement.html()), {
-          ...importedVariables, ...pageVariables, ...userDefinedVariables, ...includedVariables,
-        });
-        pageVariables[variableName] = variableValue;
-        Parser.VARIABLE_LOOKUP.get(fileName).set(variableName, variableValue);
+      const variableSource = $(this).attr('from');
+      if (variableSource !== undefined) {
+        try {
+          const variableFilePath = path.resolve(fileDir, variableSource);
+          const jsonData = fs.readFileSync(variableFilePath);
+          const varData = JSON.parse(jsonData);
+          Object.entries(varData).forEach(([varName, varValue]) => {
+            const variableValue = nunjuckUtils.renderEscaped(nunjucks, varValue, {
+              ...importedVariables, ...pageVariables, ...userDefinedVariables, ...includedVariables,
+            });
+            Parser.setPageVariable(varName, variableValue, pageVariables, fileName);
+          });
+        } catch (err) {
+          logger.warn(`Error ${err.message}`);
+        }
+      } else {
+        if (!variableName) {
+          // eslint-disable-next-line no-console
+          console.warn(`Missing 'name' for variable in ${fileName}\n`);
+          return;
+        }
+        const variableValue = nunjuckUtils.renderEscaped(
```

Comment:
Do you feel it makes sense to add this into the abstracted function too?
```
const variableValue = nunjuckUtils.renderEscaped(
          nunjucks, md.renderInline(variableElement.html()), {
            ...importedVariables, ...pageVariables, ...userDefinedVariables, ...includedVariables,
          });
```

Reply:
Or perhaps, would it make sense to define an inline function? This way we can get the benefits of abstraction while still being able to access the variables we need directly. We could do something like:
```
const processVariable(variableName, rawVariableValue) {
  const otherVariables = { ...importedVariables, ...pageVariables, ...userDefinedVariables, ...includedVariables };
  const variableValue = nunjckUtils.renderEscaped(nunjucks, rawVariableValue, otherVariables);
  if (!pageVariables[variableName]) {
      pageVariables[variableName] = variableValue;
      Parser.VARIABLE_LOOKUP.get(fileName).set(variableName, variableValue);
    }
}
```